core/test/rebin_test.cpp

```diff
@@ -96,3 +96,57 @@ TEST_F(RebinTest, keeps_unrelated_labels_but_drops_others) {
 
   ASSERT_EQ(rebin(a, Dim::X, edges), expected);
 }
+
+class RebinMask1DTest : public ::testing::Test {
+protected:
+  Variable x =
+      makeVariable<double>({Dim::X, 10}, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10});
+
+  Variable mask =
+      makeVariable<bool>({Dim::X, 10}, {false, false, true, false, false, false,
+                                        false, false, false, false});
+};
+
+TEST_F(RebinMask1DTest, mask_1d) {
+
+  const auto edges = makeVariable<double>({Dim::X, 5}, {1, 3, 5, 7, 10});
+  const auto expected =
+      makeVariable<bool>({Dim::X, 4}, {false, true, false, false});
+
+  const auto result = rebin(mask, Dim::X, x, edges);
```

Comment:
Hmmm, shouldn't this fail in a size check? The mask has the same length as the old `x` edges, but it should be 1 shorter? You have not changed that code, so it might be an existing bug?

Reply:
I did see the fail only when writing the 2D test, that's why the size of Dim::X is 6 there. I don't think the check is properly working for 1D, but it's the same code isn't it? I wondered myself why [this](https://github.com/scipp/scipp/blob/master/core/rebin.cpp#L136-L138) check wasn't failing for not bin-edges: